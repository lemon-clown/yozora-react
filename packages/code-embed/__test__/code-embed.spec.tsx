import CodeRendererJsx from '@yozora/react-code-renderer-jsx'
import type { CodeRunnerProps } from '@yozora/react-code-runners'
import { render } from 'enzyme'
import React from 'react'
import CodeEmbed from '../src'

const code = `
  function Counter() {
    const [count, setCount] = React.useState(0)
    return (
      <div>
        <button onClick={() => setCount(c => c + 1)}>+</button>
        <span>{count}</span>
        <button onClick={() => setCount(c => c - 1)}>-</button>
      </div>
    )
  }
`

const JsxRunner = ({ value }: CodeRunnerProps): React.ReactElement => {
  return (
    <CodeRendererJsx
      code={value}
      inline={true}
      onError={error => {
        console.log(error)
      }}
    />
  )
}

describe('prop types', function () {
  beforeEach(() => {
    jest.spyOn(global.console, 'error').mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })
  })

  it('value is required', function () {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(<CodeEmbed lang="jsx" value={value} runner={JsxRunner} />)
      }).toThrow(/The prop `value` is marked as required/i)
    }
  })

  describe('className is optional', function () {
    it('default', function () {
      const node = render(
        <CodeEmbed lang="jsx" value={code} runner={JsxRunner} />,
      )
      expect(node.hasClass('yozora-code-embed')).toBeTruthy()
    })

    it('custom', function () {
      const node = render(
        <CodeEmbed
          lang="jsx"
          value={code}
          runner={JsxRunner}
          className="my-code-embed"
        />,
      )
      expect(node.hasClass('yozora-code-embed')).toBeTruthy()
      expect(node.hasClass('my-code-embed')).toBeTruthy()
    })
  })

  it('style is optional', function () {
    const node = render(
      <CodeEmbed
        lang="jsx"
        value={code}
        runner={JsxRunner}
        style={{ color: 'orange' }}
      />,
    )
    expect(node.css('color')).toEqual('orange')
  })
})

describe('snapshot', function () {
  it('basic', () => {
    const wrapper = render(
      <CodeEmbed lang="jsx" value={code} runner={JsxRunner} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
