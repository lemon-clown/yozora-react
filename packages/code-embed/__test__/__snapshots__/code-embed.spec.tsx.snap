// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `
<div
  class="yozora-code-embed"
/>
`;
